apps/apps.c

```diff
@@ -2343,6 +2343,69 @@ int app_isdir(const char *name)
 }
 #endif
 
+/* app_dirname section */
+/*
+ * TODO: implement app_dirname for Windows
+ * and VMS.
+ */
+#if !defined(_WIN32) && !defined(__VMS)
+/*
+ * This exactly follows what POSIX's
+ * dirname does, but is implemented
+ * in a more platform independent way.
+ *
+ * path        dirname
+ * /usr/lib    /usr
+ * /usr/       /
+ * usr         .
+ * /           /
+ * .           .
+ * ..          .
+ * NULL        .
+ * ""          .
+ *
+ * Note: this function also keeps the
+ * possibility of modifying the 'path'
+ * string same as POSIX dirname.
+ */
+char *app_dirname(char *path)
```

Comment:
I don't really get this. You mean leave a `fallback_dirname` generally available, in case of further usage in other platforms (to parse unix file specs), such as VMS?